### ISSUE TITLE
CreateRecipe method stores a Recipe's rating as 0 instead of null if it is unrated

### ISSUE THREAD

**Issue by laurel-thomson** [dkhwang118/CIS560_F18-Project-RecipeManager#55]
When a new recipe is created its rating is assigned to 0 instead of null.  Also, the RateRecipe method is not updating the recipe table with the new rating.  Check out the Rate_Recipe_Dev branch to work on.

**Comment by laurel-thomson** [comment]
Update: I got the RateRecipe method to work.  I had to move the connection.Open() call to before the transaction using statement.

**Comment by dkhwang118** [comment]
Hi! So, if it's still storing the 0's, I think if you just add "if (recipeRating != null)" in front of the "command.Parameters.AddWithValue("RecipeRating", recipeRating);" line, it should prevent this. For some reason I cannot check this functionality in the branch because I am getting an exception thrown because of a failed ReadRecipe() lookup, but I suspect that the 0 is occurring because we're trying to pass in a value to the DB, and that even passing in a value of "null" might get converted to "DEFAULT(INT)" or something like that.

**Comment by dkhwang118** [comment]
Ok, so I just pushed some commits that fixes this issue. Though, I would like someone to double-check before I close it.

A few Notes:
-It seems that if we initialize the recipeRating variable to null, when it is passed as an argument in the AddWithValue method it acts as if we did not pass a value at all, and throws an exception saying we didn't give enough parameters to the stored procedure (which I suppose makes sense in retrospect).
-The declaration of the CreateRecipe method in IQuery had a "0" for the default value instead of "null" and I believe that was what was causing the "0" value to show up. Sorry about that :(
